### PR TITLE
196 modify dep file creation update process

### LINKED ISSUE #196: Dependency files should be created/updated when document is open or using code action
## Description ##

In order to prevent unexpected issue when autosave option is enabled, the dependency file for missing copybooks should be created and updated only in two conditions:
- the document is open in the editor and there is a copy instruction
- after the triggering from the code action

## Steps to reproduce ##
1. Verify that autosave is enabled
2. Open a cobol file
3. Start to write copy instruction taking few second of delay for each letter of the copybook name
4. Bunch of items in the dependency file will be created (one for each letter written)

## Expected behavior ##

The depfile should contain the full name of the copybook, adding each character as different copybook lead in issues and unexpected behavior.
 
### Inside the cobol file... ###
```
// other stuff
COPY ABCDEF.
// other stuff
```
### Dependency file create for the Cobol file: ###

```
A
AB
ABC
ABCD
ABCDE
ABCDEF
```

## Additional information ##

Is necessary double check the expected behavior so in that case a question tag is added.

## Technical notes ##
Update unit tests:
- [x] ```FileSystemE2ETest```
- [x] ```CobolLanguageEngineTest```
- [x] ```CobolCleanExtraLanguageTest```
- [x] ```CobolVariableCheckTest```
- [x] ```DocumentExtensionTests``` //double check later
- [x] ```MyTextDocumentServiceTest```
- [x] ```AnalyzeCopybookCaching```

### PR DESCRIPTION
**Pull request by zacanbrcom**
Closes #196
Objective of this PR is to create/update the dependency file only when the cobol file is in the DID_OPEN sync state.
As part of the review please test it on latest Che 7 (with autosave enabled) and let me know in case of unexpected behavior.
**I also updated the changelogs so please check that your contribution is there.**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,13 @@
 All notable changes to the COBOL Language Support extension are documented in this file.
 
 ## [0.x.x] - TBD
-
+- Modify dep file creation/update process:[#209](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/209) Thanks @[zacanbrcom](https://github.com/zacanbrcom)
```
Comment (temanbrcom) [inline CHANGELOG.md]: An extra colon in this line

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,13 @@
...
+- Retrieve settings.json at initialize [#207](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/207) Thanks @[sergiuilie](https://github.com/sergiuilie)
+- Introduce Code Action to Resolve Copybooks [#198](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/198) Thanks @[temanbrcom](https://github.com/temanbrcom)
+- Make CobolVisitor constructor-configurable to simplify DI [#205](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/205) Thanks @[temanbrcom](https://github.com/temanbrcom)
+- Copybook extension check fails [#203](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/203) Thanks @[grianbrcom](https://github.com/grianbrcom)
```
Comment (temanbrcom) [inline CHANGELOG.md]: To keep the naming convention, it is better to put "Fix copybook extension check"

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,13 @@
 All notable changes to the COBOL Language Support extension are documented in this file.
 
 ## [0.x.x] - TBD
-
+- Modify dep file creation/update process:[#209](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/209) Thanks @[zacanbrcom](https://github.com/zacanbrcom)
+- Retrieve settings.json at initialize [#207](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/207) Thanks @[sergiuilie](https://github.com/sergiuilie)
+- Introduce Code Action to Resolve Copybooks [#198](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/198) Thanks @[temanbrcom](https://github.com/temanbrcom)
+- Make CobolVisitor constructor-configurable to simplify DI [#205](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/205) Thanks @[temanbrcom](https://github.com/temanbrcom)
```
Comment (temanbrcom) [inline CHANGELOG.md]: I think this small refactoring should not appear in the changelog

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,7 +3,13 @@
...
+- Copybook extension check fails [#203](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/203) Thanks @[grianbrcom](https://github.com/grianbrcom)
+- Remove error listener from CobolVisitor to simpily DI [#201](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/201) Thanks @[temanbrcom](https://github.com/temanbrcom)
```
Comment (temanbrcom) [inline CHANGELOG.md]: Also here, this small refactoring should not appear in the changelog

File: clients/cobol-lsp-vscode-extension/package-lock.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "cobol-language-support",
-	"version": "0.10.0",
+	"version": "0.10.1",
```
Comment (temanbrcom) [inline clients/cobol-lsp-vscode-extension/package-lock.json]: @sergiuilie wasn't it changed during the last release?

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/engine/CobolLanguageEngine.java [hunk 6]
```diff
@@ -31,15 +31,30 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/**
+ * This class is responsible for run the syntax and semantic analysis of an input cobol document.
+ * Its run method is used by the service facade layer CobolLanguageEngineFacade
+ */
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/engine/CobolLanguageEngine.java]: 👍

File: CHANGELOG.md (hunk 3 above)
Comment (zacanbrcom) [reply CHANGELOG.md]: I already planned a small effort to review all the changelog items before the new release, we'll do it in any case also to better check if we miss smth

Comment (temanbrcom) [reply CHANGELOG.md]: Ok, I agree

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/CobolSemanticParser.java [hunk 7]
```diff
@@ -30,10 +30,14 @@
    *
```
Comment (sergiuilie) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/CobolSemanticParser.java]: the header on those files, you may skip it for now and I will change it tomorrow for the entire project

File: com.ca.lsp.cobol/lsp-core-domain/src/test/java/com/broadcom/lsp/domain/cobol/event/CopybookEventsTest.java [hunk 8]
```diff
@@ -37,6 +37,7 @@
   private static final String COBOL_FILE_URI = "file:///C:/Users/test/Main.cbl";
 
   private static final String COPYBOOK_CONTENT = "000000 IDENTIFICATION DIVISION.";
+  private static final String DID_OPEN = "DID_OPEN";
```
Comment (sergiuilie) [inline com.ca.lsp.cobol/lsp-core-domain/src/test/java/com/broadcom/lsp/domain/cobol/event/CopybookEventsTest.java]: you may edit the enum TextDocumentSyncType to put also a label and use it everywhere you need

File: clients/cobol-lsp-vscode-extension/package-lock.json (hunk 5 above)
Comment (sergiuilie) [reply clients/cobol-lsp-vscode-extension/package-lock.json]: it is package-lock, it is updated automatically, no need to commit it

**Pull request merged**